test/http-error.js

```diff
@@ -0,0 +1,46 @@
+import test from 'ava';
+import {Response} from 'node-fetch';
+import ky from '..';
+
+function createFakeResponse({status, statusText}) {
+	// Start with a realistic fetch Response.
+	const response = {...new Response()};
+
+	// Assign these values because the Response() constructor doesn't
+	// support setting them to undefined.
+	response.status = status;
+	response.statusText = statusText;
```

Comment:
These can be set in the `init` argument for the `Response` constructor. Should avoid the need to spread the response into a new object, too.

Reply:
It doesn't work because these values need to be set to `undefined` and from looking at the `node-fetch` source, they are using old school truthy checks to tell if these values are set (thus, they can never be set to `undefined`).

I realize it makes the code look a bit odd -- that's why I added a comment explaining it.